File: backend/src/main/java/com/smartstock/model/Order.java
package com.smartstock.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "purchase_order")
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne
    @JoinColumn(name = "supplier_id")
    @NotNull
    private Supplier supplier;

    @Enumerated(EnumType.STRING)
    @NotNull
    private OrderStatus status;

    @PastOrPresent
    private LocalDateTime createdAt;

    @DecimalMin(value = "0.0")
    private BigDecimal totalAmount;
}

File: frontend/src/models/order.js
/**
 * @typedef {Object} Order
 * @property {string} id
 * @property {string} supplierId
 * @property {'PENDING' | 'SHIPPED' | 'RECEIVED'} status
 * @property {string} createdAt
 * @property {number} totalAmount
 */

export const exampleOrder = {
    id: "uuid",
    supplierId: "uuid",
    status: "PENDING",
    createdAt: "2025-09-20T10:00:00Z",
    totalAmount: 25000
  };
  